bin/check-kube-external-dns-services.rb

```diff
@@ -0,0 +1,135 @@
+#     --services SERVICES          Optional list of services to check.
+#                                  Defaults to 'all'
+#
+# NOTES:
+# => The filter used for the -f flag is in the form key=value. If multiple
+#    filters need to be specfied, use a comma. ex. foo=bar,red=color
+#
+require 'resolv'
+require 'sensu-plugins-kubernetes/cli'
+require 'sensu-plugins-kubernetes/cli/namespaced'
+
+class KubeExternalDNSServices < Sensu::Plugins::Kubernetes::CLI
+  include Sensu::Plugins::Kubernetes::NamespacedCLI
+
+  option :services,
+         description: 'List of services to check',
+         long: '--services',
+         proc: proc { |a| a.split(',') },
+         default: []
+
+  option :label_filter,
+         description: 'Label selector for services to be checked (example -- key1=value1,key2!=value2)',
+         short: '-f FILTER',
+         long: '--filter'
+
+  option :exclude_namespace,
+         description: 'Exclude the specified list of namespaces',
+         short: '-n NAMESPACES',
+         long: '--exclude-namespace',
+         proc: proc { |a| a.split(',') },
+         default: ''
+
+  option :include_namespace,
+         description: 'Include the specified list of namespaces',
+         short: '-i NAMESPACES',
+         long: '--include-namespace',
+         proc: proc { |a| a.split(',') },
+         default: ''
+
+  HOSTNAME_ANNOTATION = 'external-dns.alpha.kubernetes.io/hostname'.freeze
+
+  # Returns a sorted array of external-dns hostnames set on the given service
+  def hostnames_for_service(service)
+    return [] if service.metadata.annotations.nil? || service.metadata.annotations[HOSTNAME_ANNOTATION].nil?
+    service.metadata.annotations[HOSTNAME_ANNOTATION].gsub(/\s/, '').split(',').sort
+  end
+
+  # Returns a sorted array of IPs for a given service that are exportable by external-dns
+  def addresses_for_service(service)
+    return [] if service.status.loadBalancer.nil? || service.status.loadBalancer.ingress.nil?
+    service.status.loadBalancer.ingress.map(&:ip).compact.sort
+  end
+
+  def should_exclude_namespace(namespace)
+    return !config[:include_namespace].include?(namespace) unless config[:include_namespace].empty?
+    config[:exclude_namespace].include?(namespace)
+  end
+
+  def run
+    bad_service_msg_map = {}
+
+    services = if config[:label_filter].nil?
+                 client.get_services(namespace: namespace)
+               else
+                 client.get_services(namespace: namespace, label_selector: config[:label_filter].to_s)
+               end
+
+    services.each do |s|
+      next if should_exclude_namespace(s.metadata.namespace)
+      next unless config[:services].empty? || config[:services].include?(s.metadata.name)
+
+      hostnames = hostnames_for_service(s)
+      next if hostnames.empty?
+
+      addresses = addresses_for_service(s)
+      next if addresses.empty?
+
+      msg_parts = []
+      hostnames.each do |hostname|
```

Comment:
Can we do `h` here for the iterator so its consistent with your `.each` services block above?

Reply:
Pushed.